src/engine/PolygonalTightening.h

```diff
@@ -0,0 +1,101 @@
+/*********************                                                        */
+/*! \file PolygonalTightening.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Duligur Ibeling, Guy Katz, Ido Shmuel
+ ** This file is part of the Marabou project.
+ ** Copyright (c) 2017-2024 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved. See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** [[ Add lengthier description here ]]
+
```

Comment:
Could you add a description of what this class is supposed to represent? Does this represent a linear symbolic bound in terms of neurons? Also, this file should probably be in `src/nlr/` since it's very NLR specific.